src/platform/macos/window.rs

```diff
@@ -505,13 +515,33 @@ unsafe fn get_current_monitor() -> RootMonitorId {
 
 impl Drop for Window2 {
     fn drop(&mut self) {
+        // Remove this window from the `EventLoop`s list of windows.
+        // The destructor order is:
+        // Window ->
+        // Rc<Window2> (makes Weak<..> in shared.windows none) ->
```

Comment:
It would also be nice to have `none` be `None`, for clarity. We should be good to merge after this.